From da402dd85419175aea42ce52400c4a9d402dcf49 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 24 Aug 2022 06:19:27 +0300
Subject: [PATCH] [SW-86869] gaudi3: modify polarity of recovery location fuse bit

Since the POR recovery flow is EROM its been decided to change
polarity of recovery code location fuse bit where 0 value will execute
ROM recovery and 1 will execute EROM recovery, so adjust code
accordingly.

Change-Id: Ic42de7854e4f890072fff1c46f1d0ee4619cc55e
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 572f735..f8ee5e7 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -235,10 +235,10 @@
 
 	/*
 	 * efuse[EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK]:
-	 * 0: eROM
-	 * 1: ROM
+	 * 0: ROM
+	 * 1: eROM
 	 */
-	if (!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK))
+	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK)
 		execute_erom_recovery();
 	else
 		execute_rom_recovery();
